code/game/objects/items/implants/implant_explosive.dm

```diff
@@ -61,15 +61,16 @@
 	timed_explosion()
 
 /obj/item/implant/explosive/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
-	for(var/X in target.implants)
-		if(istype(X, /obj/item/implant/explosive)) //we don't use our own type here, because macrobombs inherit this proc and need to be able to upgrade microbombs
-			var/obj/item/implant/explosive/imp_e = X
-			imp_e.heavy += heavy
-			imp_e.medium += medium
-			imp_e.weak += weak
-			imp_e.delay += delay
-			qdel(src)
-			return TRUE
+	for(var/implant as anything in target.implants)
```

Comment:
AFAIK, you don't need an `as anything` in a loop which doesn't cast to any type immediately